### PR TITLE
use double indent for function parameters

### PR DESCRIPTION
**Pull request by kevinushey**
This PR modifies our indentation rules to support https://github.com/tidyverse/style/pull/174/files.

https://user-images.githubusercontent.com/1976582/130520726-9d4ca8a1-a9c8-45a8-9f73-83466c51cfb5.mov

cc: @hadley

### CODE REVIEW

File: src/gwt/acesupport/acemode/rmarkdown.js [hunk 1]
```diff
@@ -193,6 +193,16 @@ oop.inherits(Mode, MarkdownMode);
          return this.$outdent.autoOutdent(session, row);
    };
 
+   this.getIndentForOpenBrace = function(openBracePos)
```
Comment (kevinushey) [inline src/gwt/acesupport/acemode/rmarkdown.js:196]: This definition is necessary to ensure that the closing `}` is formatted correctly when inserting a newline as in e.g.

```
foo <- function() {|}
```

with `|` denoting cursor position.

**Comment by jmcphers** [comment]
@hadley Were there any objections?

**Comment by hadley** [comment]
@jmcphers sorry, we realised that we were talking past each other so there's no as much consensus as I thought. And then I dropped the ball on following up 😞. I've re-added to my to do list.

**Comment by jmcphers** [comment]
Okay, thanks. I'm converting this to a draft so it doesn't accidentally get merged until then. (Although if consensus was low enough, maybe we should just close it?)

**Comment by hadley** [comment]
I think it's still worth doing; I just need to double check. (The discussion might also be more about what the style guide says vs what RStudio does)

**Comment by hadley** [comment]
I've been convinced that I was being too stringent by only allowing double-indent style so I've updated the PR to allow either function- or double-indent style depending on the overall width.  

I think this means that this PR can be merged — it would be cool if it could somehow conditionally support double- or function-indent style (using some heuristic based on the column position), but that seems like a lot of work, and this PR is still an improvement because it pushes people towards one of the supported styles.

Either way, we're likely to work on an add-in that makes it easier to toggle between the different styles of function indenting.

**Comment by jmcphers** [comment]
Okay, will merge as is. Thanks for following up!

**Pull request merged**